src/importlinter/contracts/layers.py

```diff
@@ -173,6 +187,28 @@ def _check_all_layers_exist_for_container(self, container: str, graph: ImportGra
                     f"module {layer_module_name} does not exist."
                 )
 
+    def _check_all_potential_layers_exist_in_contract(
+        self, container: str, graph: ImportGraph
+    ) -> None:
+        declared_layers = set(layer.name for layer in self.layers)  # type: ignore
+        expected_layers = declared_layers | self.exhaustive_ignores  # type: ignore
```

Comment:
why `# type: ignore` is needed here?

Reply:
In these specific cases, the error in the first line is that you cannot iterate `self.layers` and the second is that you cannot `|` between `Set` and `SetField`.

More generaly, these ignores are needed throughout the contract modules due to the way that the `Field` subclasses are dynamically replaced with their parsed values upon class instantiation. This could be worked around by adding a stub file for `fields.py` that uses a descriptor protocol when declaring `Field`, but that seemed like overkill for this PR.